server-spi-private/src/main/java/org/keycloak/models/LoginFailureSpi.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.models;
+
+import org.keycloak.provider.Provider;
+import org.keycloak.provider.ProviderFactory;
+import org.keycloak.provider.Spi;
+
+/**
+ * @author <a href="mailto:mkanis@redhat.com">Martin Kanis</a>
+ */
+public class LoginFailureSpi implements Spi {
+
+    public static final String NAME = "loginFailures";
```

Comment:
That would be good even though it works correctly now. Please use singular for SPI name rather than plural, just like in case of _realm_, _client_ or _group_. There should be a cleanup eventually to make this consistent across other introduced spis that started using plural form.

Reply:
Done. `loginFailures` changed to singular in the SPI and `keycloak-server.json`